src/Registry.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+error AlreadyAdded();
+
+/// @title  Registry contract :: Adding popular token's data :: DAO Owned 
+/// @notice This contract can be called by our main contract for fetching tokenData
+
+contract Registry {
+
+    ///  Address of the DAO
+    address immutable owner_DAO;
+
+    /// This struct contains necessary info about an ERC20 token
+    /// What more params could be added?
+    struct Token{
+        string name;
+        string symbol;
+        uint8 UID;
+        uint8 decimals;
+    }
+    
+    /// Registry mapping ERC20 token address to its metadata
+    mapping(address=>Token) tokenData;
+    mapping(address=>bool)  isLegit;
+
+    /// All functions to be called only by the DAO
+    modifier isOwner() {
+        require(msg.sender==owner_DAO);
+        _;
+    }
+  
+    /// msg.sender should be a DAO controlled address
+    constructor(){
+        owner_DAO = msg.sender;  
+    }
```

Comment:
let's use OpenZeppelin Ownable instead. It's a standard that's supported by other tools and it allows us to transfer or renounce ownership eventually